src/lib/efhw/efct.c

```diff
@@ -61,17 +64,29 @@ efct_nic_rxq_bind(struct efhw_nic *nic, int qid, bool timestamp_req,
     .n_hugepages = n_hugepages,
   };
 
-  if( n_hugepages > CI_EFCT_MAX_HUGEPAGES ) {
+  /* We implicitly lock here by calling `efct_provide_hugetlb_alloc` so that
+   * `used_hugepages` does not become invalid between now and binding */
+  efct_provide_hugetlb_alloc(hugetlb_alloc);
+  used_hugepages = efct_get_used_hugepages(nic, qid);
+  if( used_hugepages < 0 ) {
+    efct_unprovide_hugetlb_alloc();
+    return used_hugepages;
+  }
+
+  EFHW_ASSERT(used_hugepages <= CI_EFCT_MAX_HUGEPAGES);
+
+  if( n_hugepages + used_hugepages > CI_EFCT_MAX_HUGEPAGES ) {
     /* We'd fail later on in bind_rxq(), but only after we'd done a load of
      * shuffling-about and memory allocation. So let's have this early sanity
      * check as well */
```

Comment:
Didn't we add this code because this comment is no longer completely true?

Reply:
I think we added this code so we don't end up donating superbufs to the pool such that an sbid `>= CI_EFCT_MAX_SUPERBUFS` isn't returned. I'll update the comment.